File: fixtures/csharp/type_inference/overloads/overloads.cs
namespace OverloadApp
{
    public class Overloads
    {
        public int Combine(int a, int b)
        {
            return a + b;
        }

        public string Combine(string a, string b)
        {
            return a + b;
        }

        public string Combine(int a, string b)
        {
            return a + b;
        }

        public string Run()
        {
            int ints = Combine(1, 2);
            string strs = Combine("a", "b");
            string mixed = Combine(1, "b");
            return $"{ints}{strs}{mixed}";
        }
    }
}

File: fixtures/cpp/type_inference/overloads/overloads.cpp
#include <string>
#include <sstream>

class Overloads {
public:
    int combine(int a, int b) {
        return a + b;
    }

    std::string combine(const std::string& a, const std::string& b) {
        return a + b;
    }

    std::string combine(int a, const std::string& b) {
        std::ostringstream oss;
        oss << a << b;
        return oss.str();
    }

    std::string run() {
        int ints = combine(1, 2);
        std::string strs = combine("a", "b");
        std::string mixed = combine(1, "b");
        std::ostringstream oss;
        oss << ints << strs << mixed;
        return oss.str();
    }
};
